### PR TITLE
Docker for test AutoUpgrade

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Separate services in multiple docker-compose files to be capable to use different docker files for differen versions of PS
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | To test upgrade from 1.7.5.2 to 1.7.6.0 : <br> Cmd1: `PS_VERSION_TO_INSTALL=1.7.5.2 docker-compose -f docker-compose.ps-git.yml -f docker-compose.tests.yml up --build` <br> Cmd2: `docker-compose -f docker-compose.tests.yml -f docker-compose.ps-git.yml exec -e COMMAND="specific-test" -e PS_VERSION="1.7.6.0" -e TEST_PATH="upgrade/upgradeShop" tests bash /tmp/run-tests.sh`

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/15163)

### CODE REVIEW

**Comment by boubkerbribri** [comment]
WIP, please do not merge yet.

File: tests/puppeteer/.docker/prestashop/ps_v175
Comment (PierreRambaud) [inline tests/puppeteer/.docker/prestashop/ps_v175]:
```suggestion
ENV PS_VERSION 1.7.6.0
```

Comment (PierreRambaud) [inline tests/puppeteer/.docker/prestashop/ps_v175]: Could we use the current working repository instead?

File: tests/puppeteer/.docker/prestashop/ps_v175
Comment (PierreRambaud) [inline tests/puppeteer/.docker/prestashop/ps_v175]: Could we use https://hub.docker.com/r/prestashop/prestashop-git and make it configurable?

Comment (PierreRambaud) [reply tests/puppeteer/.docker/prestashop/ps_v175]: and we will be able to do:
```bash
docker-compose -e PS_VERSION=1.7.5.2 -f docker-compose.autoupgrade.yml -f docker-compose.tests.yml up --build
```

Comment (boubkerbribri) [reply tests/puppeteer/.docker/prestashop/ps_v175]: done

**Comment by SimonGrn** [comment]
Do you think it's possible to change the README as well ?

**Comment by boubkerbribri** [comment]
> Do you think it's possible to change the README as well ?

Yes, I'm doing it (that's why PR is still on WIP)

File: tests/puppeteer/.docker/prestashop/docker_run.sh
Comment (PierreRambaud) [inline tests/puppeteer/.docker/prestashop/docker_run.sh]: You can add `set -x` as the beginning of the script if you want to print a trace of simple commands

File: tests/puppeteer/docker-compose.ps-git.yml
Comment (PierreRambaud) [inline tests/puppeteer/docker-compose.ps-git.yml]:
```suggestion
      build: .docker/prestashop
```

File: tests/puppeteer/.docker/prestashop/ps_git
Comment (PierreRambaud) [inline tests/puppeteer/.docker/prestashop/ps_git]: rename in Dockerfile

File: tests/puppeteer/.docker/prestashop/docker_run.sh
Comment (PierreRambaud) [reply tests/puppeteer/.docker/prestashop/docker_run.sh]: Rename this file in `docker_run_git.sh`
And at the end of this file, run `/tmp/docker_run.sh`.
At least, if the file change in the prestashop/prestashop-git` image, you don't have to edit this file.

Comment (boubkerbribri) [reply tests/puppeteer/.docker/prestashop/docker_run.sh]: there is already a file named `docker_run_git.sh` (imported by the image `prestashop-git` and where we run Composer install and call `docker_run.sh` after)

File: tests/puppeteer/.docker/prestashop/ps_git
Comment (boubkerbribri) [reply tests/puppeteer/.docker/prestashop/ps_git]: Do mean to have only one DockerFile or to just change extension ? (because we already have a Dockerfile used to import local version to docker container)

Comment (PierreRambaud) [reply tests/puppeteer/.docker/prestashop/ps_git]: as we said in private. juste rename in `Dockerfile-from-git` or something similar

File: tests/puppeteer/.docker/prestashop/docker_run.sh
Comment (PierreRambaud) [reply tests/puppeteer/.docker/prestashop/docker_run.sh]: Oh, didn't notice that. Thanks =)

**Review by SimonGrn — DISMISSED** [review]
Thanks @boubkerbribri

File: tests/puppeteer/.docker/prestashop/ps_git
Comment (boubkerbribri) [reply tests/puppeteer/.docker/prestashop/ps_git]: Done thanks

**Comment by boubkerbribri** [comment]
Please test it before merging it.
Thanks

**Review by PierreRambaud — DISMISSED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Pull request merged**
